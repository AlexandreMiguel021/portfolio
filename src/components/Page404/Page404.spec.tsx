import { render, screen } from '@testing-library/react'
import { Page404 } from '.'
import '@testing-library/jest-dom'

describe('Page404', () => {
	it('should render a Page404 correctly', () => {
		const { container } = render(<Page404 />)
		expect(container).toMatchSnapshot()
	})

	it('should render a page404 with info error text', () => {
		render(<Page404 />)
		expect(screen.getByText('Página não encontrada!')).toBeInTheDocument()
	})

	it('should render a page404 with link to return homepage', () => {
		const { container } = render(<Page404 />)
		expect(screen.getByText('Voltar')).toBeInTheDocument()
		expect(container.querySelector('a')).toHaveAttribute('href', '/')
		expect(container.querySelector('a')).toHaveAttribute(
			'aria-label',
			'voltar para página principal'
		)
	})
})
