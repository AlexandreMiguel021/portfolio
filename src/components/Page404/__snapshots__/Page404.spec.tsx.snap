// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page404 should render a Page404 correctly 1`] = `
<div>
  <div
    class="wrapper"
  >
    <h1>
      404
    </h1>
    <p>
      Página não encontrada!
    </p>
    <a
      aria-label="voltar para página principal"
      class="component large"
      href="/"
      target="_self"
    >
      <span
        class="icon"
      >
        <svg
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 8a.5.5 0 0 0-.5-.5H2.707l3.147-3.146a.5.5 0 1 0-.708-.708l-4 4a.5.5 0 0 0 0 .708l4 4a.5.5 0 0 0 .708-.708L2.707 8.5H14.5A.5.5 0 0 0 15 8z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <span
        class="text"
      >
        Voltar
      </span>
    </a>
  </div>
</div>
`;
